SPECS/prometheus-node-exporter/prometheus-node-exporter.spec

```diff
@@ -0,0 +1,139 @@
+License:        ASL 2.0 AND MIT
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            https://github.com/prometheus/node_exporter
+Source0:        https://github.com/prometheus/node_exporter/archive/refs/tags/v%{version}.tar.gz#/%{name}-%{version}.tar.gz
+# Below is a manually created tarball, no download link.
+# We're using vendored Go modules from this tarball, since network is disabled during build time.
+# How to re-build this file:
+#   1. wget https://github.com/prometheus/node_exporter/archive/refs/tags/v%{version}.tar.gz -O %%{name}-%%{version}.tar.gz
+#   2. tar -xf %%{name}-%%{version}.tar.gz
+#   3. cd %%{name}-%%{version}
+#   4. Apply patches from the spec (may change go dependencies).
+#   5. go mod vendor
+#   6. tar  --sort=name \
+#           --mtime="2021-04-26 00:00Z" \
+#           --owner=0 --group=0 --numeric-owner \
+#           --pax-option=exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime \
+#           -cf %%{name}-%%{version}-vendor.tar.gz vendor
+#
+Source1:        %{name}-%{version}-vendor.tar.gz
+Source2:        %{name}.sysusers
+Source3:        %{name}.service
+Source4:        %{name}.conf
+Source5:        %{name}.logrotate
+# Replace defaults paths for config files
+Patch0:         defaults-paths.patch
+# https://github.com/prometheus/node_exporter/pull/2190
+Patch1:         0001-Refactor-perf-collector.patch
+
+BuildRequires:  golang
+BuildRequires:  systemd-rpm-macros
+
+Requires(pre):  shadow-utils
+
+%description
+Prometheus exporter for hardware and OS metrics exposed by *NIX kernels, written
+in Go with pluggable metric collectors.
+
+%prep
+%autosetup -p1 -n node_exporter-%{version}
+
+rm -rf vendor
+tar -xf %{SOURCE1} --no-same-owner
+
+%build
+export BUILDTAGS="netgo osusergo static_build"
+LDFLAGS="-X github.com/prometheus/common/version.Version=%{version}      \
+         -X github.com/prometheus/common/version.Revision=%{release}     \
+         -X github.com/prometheus/common/version.Branch=tarball          \
+         -X github.com/prometheus/common/version.BuildDate=%{build_date} \
+         -X github.com/ncabatoff/process-exporter/version.GoVersion=%{go_version}"
+go build -ldflags "$LDFLAGS" -mod=vendor -v -a -tags "$BUILDTAGS" -o bin/node_exporter ./collector
+
+%install
+install -m 0755 -vd %{buildroot}%{_bindir}
+install -m 0755 -vp bin/* %{buildroot}%{_bindir}/
+mv %{buildroot}%{_bindir}/node_exporter %{buildroot}%{_bindir}/%{name}
+ln -s %{name} %{buildroot}%{_bindir}/node_exporter
+
+install -Dpm0644 %{SOURCE2} %{buildroot}%{_sysusersdir}/%{name}.conf
+install -Dpm0644 %{SOURCE3} %{buildroot}%{_unitdir}/%{name}.service
+install -Dpm0644 %{SOURCE4} %{buildroot}%{_sysconfdir}/default/%{name}
+install -Dpm0644 example-rules.yml %{buildroot}%{_datadir}/prometheus/node-exporter/example-rules.yml
+install -Dpm0644 %{SOURCE5} %{buildroot}%{_sysconfdir}/logrotate.d/%{name}
+mkdir -vp %{buildroot}%{_sharedstatedir}/prometheus/node-exporter
+
+%check
+make test
```

Comment:
Wondering if the `make test` respects the `vendor` folder or tries to fetch stuff from the internet ?

Reply:
I've sent you a message with some details offline.:)